internal/core/CMakeLists.txt

```diff
@@ -39,6 +39,14 @@ message( STATUS "Building using CMake version: ${CMAKE_VERSION}" )
 project(core)
 include(CheckCXXCompilerFlag)
 if ( APPLE )
+    message("YX: CMAKE_C_COMPILER: ${CMAKE_C_COMPILER}")
+    message("YX: CMAKE_CXX_COMPILER: ${CMAKE_CXX_COMPILER}")
+    if ( APPLE )
+        set( CMAKE_CROSSCOMPILING TRUE )
+        set( RUN_HAVE_GNU_POSIX_REGEX 0 )
+        set( CMAKE_C_COMPILER "/usr/local/opt/llvm/bin/clang" )
+        set( CMAKE_CXX_COMPILER "/usr/local/opt/llvm/bin/clang++" )
+    endif ()
```

Comment:
execute_process (
        COMMAND bash -c "brew --prefix llvm"
        OUTPUT_VARIABLE LLVM_PREFIX
    )

    set( CMAKE_C_COMPILER "/usr/local/opt/llvm/bin/clang" )
    message( STATUS "Check clang set to ${LLVM_PREFIX}/bin/clang")
    set( CMAKE_CXX_COMPILER "/usr/local/opt/llvm/bin/clang++" )
    message( STATUS "Check clang not exist, set to ${LLVM_PREFIX}/bin/clang++")